### PR TITLE
Adding deploying vm with GPU option

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Adding deploying vm with GPU option
### Changes

Adding gpu list to the vm
### Related Issues

#165

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstring

### CODE REVIEW

**Review by rawdaGastan — CHANGES_REQUESTED** [review]
we should update the docs

File: grid-cli/docs/vm.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ tf-grid-cli deploy vm [flags]
 - memory: memory size in GB (default 1).
 - rootfs: root filesystem size in GB (default 2).
 - ygg: assign yggdrasil ip for VM (default true).
+- gpus: assign alist of gpus to VM. note: setting this without the node option will fail.
```
Comment (xmonader) [inline grid-cli/docs/vm.md]: alist -> a list, but i'm not sure how it will look still

File: grid-cli/cmd/deploy_vm.go [hunk 2]
```diff
@@ -12,11 +12,19 @@ import (
 	"github.com/threefoldtech/tfgrid-sdk-go/grid-cli/internal/filters"
 	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/deployer"
 	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/workloads"
+	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
 )
 
 var ubuntuFlist = "https://hub.grid.tf/tf-official-apps/threefoldtech-ubuntu-22.04.flist"
 var ubuntuFlistEntrypoint = "/sbin/zinit init"
 
+func convertGPUsToZosGPUs(gpus []string) (zosGPUs []zos.GPU) {
```
Comment (mariobassem) [inline grid-cli/cmd/deploy_vm.go:21]: i believe you need to upgrade zos version to have the GPU types

**Review by mariobassem — CHANGES_REQUESTED** [review]
i believe you need to update zos version

File: grid-cli/cmd/deploy_vm.go (hunk 2 above)
Comment (rawdaGastan) [reply grid-cli/cmd/deploy_vm.go:21]: it will wait until client pr got merged

File: grid-cli/docs/vm.md (hunk 1 above)
Comment (maayarosama) [reply grid-cli/docs/vm.md]: Should be something like this
```bash
tf-grid-cli deploy vm --name examplevm --ssh ~/.ssh/id_rsa.pub --cpu 2 --memory 4 --disk 10 --gpus gpu1 --gpus gpu2 --node 12```

**Review by rawdaGastan — APPROVED** [review]
tested 👍🏻

File: grid-cli/docs/vm.md (hunk 1 above)
Comment (xmonader) [reply grid-cli/docs/vm.md]: is it the `id` string?

Comment (rawdaGastan) [reply grid-cli/docs/vm.md]: can be added to the docs (gpus IDs)?

**Review by xmonader — APPROVED** [review]

**Pull request merged**